### ISSUE TITLE
Issue with ogc:And

### ISSUE THREAD

**Issue by 1Map** [highsource/ogc-schemas#75]
Hi, I am trying to implement your source with a online SLD editor that I am designing. I get the following message though:

Element [{http://www.opengis.net/ogc}ogc:And] is not known in this context and property does not allow DOM.

My XML is as follow:

```xml
<?xml version="1.0" encoding="ISO-8859-1"?>
<StyledLayerDescriptor 
  version="1.0.0" 
  xmlns="http://www.opengis.net/sld" 
  xmlns:ogc="http://www.opengis.net/ogc"
  xmlns:xlink="http://www.w3.org/1999/xlink" 
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://www.opengis.net/sld http://schemas.opengis.net/sld/1.0.0/StyledLayerDescriptor.xsd">
  <NamedLayer>
    <Name>Custom Style</Name>
    <UserStyle>
      <Title>Default custom style</Title>
      <Abstract>A sample style</Abstract>
      <FeatureTypeStyle>
        <Rule>
          <Name>Polygon</Name>
          <Title>Polygon Rule</Title>
          <Abstract>Default Style for Polygons</Abstract>
          <ogc:Filter>
            <ogc:And>
              <ogc:PropertyIsEqualTo>
                <ogc:Function name="in2">
                  <ogc:Function name="geometryType">
                      <ogc:PropertyName>the_geom</ogc:PropertyName>
                  </ogc:Function>
                  <ogc:Literal>Polygon</ogc:Literal>
                  <ogc:Literal>MultiPolygon</ogc:Literal>
                </ogc:Function>
                <ogc:Literal>true</ogc:Literal>
              </ogc:PropertyIsEqualTo>
              <ogc:PropertyIsEqualTo>
                <ogc:PropertyName>onemap_geom_valid</ogc:PropertyName>
                <ogc:Literal>true</ogc:Literal>
              </ogc:PropertyIsEqualTo>
            </ogc:And>
          </ogc:Filter>
          <MinScaleDenominator>0.0</MinScaleDenominator>
          <MaxScaleDenominator>32000.0</MaxScaleDenominator>
          <PolygonSymbolizer>
            <Fill>
              <CssParameter name="fill">#AAAAAA</CssParameter>
            </Fill>
            <Stroke>
              <CssParameter name="stroke">#000000</CssParameter>
              <CssParameter name="stroke-width">1</CssParameter>
            </Stroke>
          </PolygonSymbolizer>
        </Rule>
        <Rule>
          <Name>Point</Name>
          <Title>Point Rule</Title>
          <Abstract>Default Style for Points</Abstract>
          <ogc:Filter>
            <ogc:And>
              <ogc:PropertyIsEqualTo>
                <ogc:Function name="in2">
                  <ogc:Function name="geometryType">
                    <ogc:PropertyName>the_geom</ogc:PropertyName>
                  </ogc:Function>
                  <ogc:Literal>Point</ogc:Literal>
                  <ogc:Literal>MultiPoint</ogc:Literal>
                </ogc:Function>
                <ogc:Literal>true</ogc:Literal>
              </ogc:PropertyIsEqualTo>
              <ogc:PropertyIsEqualTo>
                <ogc:PropertyName>onemap_geom_valid</ogc:PropertyName>
                <ogc:Literal>true</ogc:Literal>
              </ogc:PropertyIsEqualTo>
            </ogc:And>
          </ogc:Filter>
          <MinScaleDenominator>0.0</MinScaleDenominator>
          <MaxScaleDenominator>32000.0</MaxScaleDenominator>
          <PointSymbolizer>
            <Graphic>
              <Mark>
                <WellKnownName>square</WellKnownName>
                <Fill>
                  <CssParameter name="fill">#FF0000</CssParameter>
                </Fill>
              </Mark>
              <Size>6</Size>
            </Graphic>
          </PointSymbolizer>
        </Rule>
        <Rule>
          <Name>Line</Name>
          <Title>Line Rule</Title>
          <Abstract>Default Style for Lines</Abstract>
          <ogc:Filter>
            <ogc:And>
              <ogc:PropertyIsEqualTo>
                <ogc:Function name="in3">
                  <ogc:Function name="geometryType">
                    <ogc:PropertyName>the_geom</ogc:PropertyName>
                  </ogc:Function>
                  <ogc:Literal>LineString</ogc:Literal>
                  <ogc:Literal>LinearRing</ogc:Literal>
                  <ogc:Literal>MultiLineString</ogc:Literal>
                </ogc:Function>
                <ogc:Literal>true</ogc:Literal>
              </ogc:PropertyIsEqualTo>
              <ogc:PropertyIsEqualTo>
                <ogc:PropertyName>onemap_geom_valid</ogc:PropertyName>
                <ogc:Literal>true</ogc:Literal>
              </ogc:PropertyIsEqualTo>
            </ogc:And>
          </ogc:Filter>
          <MinScaleDenominator>0.0</MinScaleDenominator>
          <MaxScaleDenominator>32000.0</MaxScaleDenominator>
          <LineSymbolizer>
            <Stroke>
              <CssParameter name="stroke">#0000FF</CssParameter>
              <CssParameter name="stroke-width">4</CssParameter>
            </Stroke>
          </LineSymbolizer>
        </Rule>
      </FeatureTypeStyle>
    </UserStyle>
  </NamedLayer>
</StyledLayerDescriptor>
```

**Comment by bartvde** [comment]
I have no trouble deserialising and serialising your XML with:

```
<html>
<body>
<script src="bower_components/jsonix/dist/Jsonix-all.js"></script>
<script src="bower_components/w3c-schemas/scripts/lib/XLink_1_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/OWS_1_1_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/Filter_1_1_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/GML_3_1_1.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/SMIL_2_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/SMIL_2_0_Language.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/OWS_1_0_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/WFS_1_1_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/WPS_1_0_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/SLD_1_0_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/Filter_1_0_0.js"></script>
<script src="bower_components/ogc-schemas/scripts/lib/GML_2_1_2.js"></script>
<script>
var context = new Jsonix.Context([XLink_1_0, Filter_1_0_0, GML_2_1_2, SLD_1_0_0]);
var unmarshaller = context.createUnmarshaller();
var marshaller = context.createMarshaller();
var url = 'sld.xml';
var xmlhttp = new XMLHttpRequest();
xmlhttp.open("GET", url, true);
xmlhttp.onload = function() {
    if (this.responseXML !== null) {
      var info = unmarshaller.unmarshalDocument(this.responseXML);
      window.console.log(info);
      window.console.log(marshaller.marshalString(info));
   }
};
xmlhttp.send();
</script>
</body>
</html>
```

**Comment by 1Map** [comment]
Hi bartvde,

Thanks, I use exactly the same code, except:

var sldResult = unmarshaller.unmarshalString(sldText);

And that gives me following error message:

Element [{http://www.opengis.net/ogc}ogc:And] is not known in this context and property does not allow DOM.

**Comment by bartvde** [comment]
okay works for me with that as well, might be different versions used here

```
var info = unmarshaller.unmarshalString(this.responseText);
```

```
Barts-MacBook-Air:sld-test bartvandeneijnden$ bower list
bower check-new     Checking for new versions of the project dependencies..
sld-test#0.1.0 /Users/bartvandeneijnden/opengeo/git/sld-test
├── jquery#1.11.1 (latest is 2.1.3)
├── jsonix#e285e82fb1
├── ogc-schemas#b3e213c564
└── w3c-schemas#523354e203
```

**Comment by highsource** [comment]
If you're using the latest mappings, there's #74 bug. I've already fixed it but did not commit yet. Try one of the previous (released) versions.

**Comment by 1Map** [comment]
bower install ogc-schemas

gives me

bower EMALFORMED    Failed to read /tmp/schalk/bower/ogc-schemas-23021-SHErTi/bower.json

**Comment by highsource** [comment]
Hm, I thought this was fixed in #67 and #61, but I'll check. Please file an extra issue.

**Comment by 1Map** [comment]
Closing this for now as it is part of https://github.com/highsource/ogc-schemas/issues/74
